www/js/metrics.js

```diff
@@ -699,77 +702,91 @@ angular.module('emission.main.metrics',['nvd3', 'emission.services', 'ionic-date
-                    + (twoWeeksAgoCarbonInt[0] + twoWeeksAgoCarbonInt[1]))
-      var calculation = (((lastWeekCarbonInt[0] + lastWeekCarbonInt[1]) / 2)
-                        / ((twoWeeksAgoCarbonInt[0] + twoWeeksAgoCarbonInt[1]) / 2))
-                        * 100 - 100;
+      console.log("Running calculation with " + lastWeekCarbonInt + " and " + twoWeeksAgoCarbonInt);
+      var calculation = (lastWeekCarbonInt/twoWeeksAgoCarbonInt) * 100 - 100;
 
       // TODO: Refactor this so that we can filter out bad values ahead of time
       // instead of having to work around it here
       if (isValidNumber(calculation)) {
-          if(lastWeekCarbonInt[0] > twoWeeksAgoCarbonInt[0]){
-            $scope.carbonData.change = " increase over a week";
-            $scope.carbonUp = true;
-            $scope.carbonDown = false;
-          } else {
-            $scope.carbonData.change = " decrease over a week"
-            $scope.carbonUp = false;
-            $scope.carbonDown = true;
-          }
-          $scope.carbonData.changeInPercentage = Math.abs(Math.round(calculation)) + "%"
+        if(lastWeekCarbonInt > twoWeeksAgoCarbonInt){
+          $scope.carbonData.change = " increase over a week";
+          $scope.carbonUp = true;
+          $scope.carbonDown = false;
+        } else {
+          $scope.carbonData.change = " decrease over a week"
+          $scope.carbonUp = false;
+          $scope.carbonDown = true;
+        }
+        $scope.carbonData.changeInPercentage = Math.abs(Math.round(calculation)) + "%"
+      }
+      else {
+        console.debug("TIAGO: fillFootprintCardUserVals() WARNING: calculation for carbon change is not a valid number: " + calculation);
+        $scope.carbonData.change = " (no change)";
+        $scope.carbonData.changeInPercentage = '0%';
       }
    };
 
    $scope.fillFootprintAggVals = function(aggDistance) {
       if (aggDistance) {
         var aggrCarbonData = getAvgSummaryDataRaw(aggDistance, 'distance');
+
+        // TIAGO: please remove before PR, and turn into an issue?
+        // It can happen that the mode "ON_FOOT" has a NaN value.
+        // This may be due to the fact that the aggregated values are requested
+        // from the main e-mission server and not from our own.
         for (var i in aggrCarbonData) {
-          if (aggrCarbonData[i].key === "IN_VEHICLE") {
-            $scope.carbonData.aggrVehicleRange = FootprintHelper.getFootprintRaw(aggrCarbonData[i].values, aggrCarbonData[i].key);
-            $scope.carbonData.aggrCarbon = FootprintHelper.getFootprint(aggrCarbonData[i].values, aggrCarbonData[i].key);
+          if (isNaN(aggrCarbonData[i].values)) {
+            console.debug("TIAGO: fillFootprintAggVals() WARNING: value is NaN for mode " + aggrCarbonData[i].key + ", changing to 0");
+            aggrCarbonData[i].values = 0;
           }
         }
+
+        $scope.carbonData.aggrVehicleRange = FootprintHelper.getFootprintForMetrics(aggrCarbonData);
```

Comment:
Not your change, but I don't see this variable being used anywhere.
So you could remove it, since `aggrCarbon` seems to represent the same value anyway.

Reply:
If not should probably rename to indicate that it is not a range any more.

Reply:
Removed :)